front/components/agent_builder/capabilities/knowledge/AddIncludeDataSheet.tsx

```diff
@@ -0,0 +1,166 @@
+import type { MultiPageSheetPage } from "@dust-tt/sparkle";
+import {
+  ActionIncludeIcon,
+  MultiPageSheet,
+  MultiPageSheetContent,
+} from "@dust-tt/sparkle";
+import { useCallback, useEffect, useMemo, useState } from "react";
+
+import { DataSourceSelectionPage } from "@app/components/agent_builder/capabilities/knowledge/shared/DataSourceSelectionPage";
+import { DescriptionSection } from "@app/components/agent_builder/capabilities/knowledge/shared/DescriptionSection";
+import {
+  getDataSourceConfigurations,
+  getTimeFrame,
+  hasDataSourceSelections,
+  isValidPage,
+} from "@app/components/agent_builder/capabilities/knowledge/shared/sheetUtils";
+import { TimeFrameSection } from "@app/components/agent_builder/capabilities/knowledge/shared/TimeFrameSection";
+import type {
+  AgentBuilderAction,
+  IncludeDataAgentBuilderAction,
+} from "@app/components/agent_builder/types";
+import type {
+  DataSourceViewSelectionConfigurations,
+  TimeFrame,
+} from "@app/types";
+
+const PAGE_IDS = {
+  DATA_SOURCE_SELECTION: "data-source-selection",
+  CONFIGURATION: "configuration",
+} as const;
+
+type PageId = (typeof PAGE_IDS)[keyof typeof PAGE_IDS];
+
+interface AddIncludeDataSheetProps {
+  onSave: (action: AgentBuilderAction) => void;
+  isOpen: boolean;
+  onClose: () => void;
+  action?: AgentBuilderAction;
+}
+
+export function AddIncludeDataSheet({
+  onSave,
+  isOpen,
+  onClose,
+  action,
+}: AddIncludeDataSheetProps) {
+  const [currentPageId, setCurrentPageId] = useState<PageId>(
+    PAGE_IDS.DATA_SOURCE_SELECTION
+  );
+  const [description, setDescription] = useState(action?.description ?? "");
+  const [dataSourceConfigurations, setDataSourceConfigurations] =
+    useState<DataSourceViewSelectionConfigurations>(() =>
+      getDataSourceConfigurations(action)
+    );
+  const [timeFrame, setTimeFrame] = useState<TimeFrame | null>(() =>
+    getTimeFrame(action)
+  );
+
+  useEffect(() => {
```

Comment:
Do we have to have this because we don't unmount this component? and we don't want to unmount because we cannot add exit animation? Can we at least set the state only when isOpen is true?

Reply:
Can we actually set this on onOpenChange? 🤔

Reply:
> Do we have to have this because we don't unmount this component? and we don't want to unmount because we cannot add exit animation?

Exactly

> Can we actually set this on onOpenChange? 🤔

Hmmm I actually went for the useEffect option because I'm not able to make it work properly with onOpenChange